app/src/main/java/org/septa/android/app/notifications/PushNotificationManager.java

```diff
@@ -0,0 +1,502 @@
+
+        NotificationCompat.Builder mBuilder = new NotificationCompat.Builder(context, CHANNEL_ID)
+                .setSmallIcon(R.drawable.ic_launcher)
+                .setContentTitle(title)
+                .setContentText(message)
+                .setStyle(new NotificationCompat.BigTextStyle().bigText(message));
+
+        // estimated delay notifications don't link anywhere in app, actual ones do
+        if (delayType == DelayNotificationType.ACTUAL) {
+
+            // look up destination stop
+            StopModel destStop = getRailStop(context, destinationStopId);
+
+            // look up routeName
+            String routeName = getRailRouteName(context, routeId);
+
+            // build intent for click action
+            final Intent resultIntent = new Intent(context, MainActivity.class);
+            resultIntent.putExtra(Constants.REQUEST_CODE, Constants.PUSH_NOTIF_REQUEST_RAIL_DELAY);
+            resultIntent.putExtra(Constants.DESTINATION_STATION, destStop);
+            resultIntent.putExtra(Constants.DESTINATION_STOP_ID, destinationStopId);
+            resultIntent.putExtra(Constants.VEHICLE_ID, vehicleId);
+            resultIntent.putExtra(Constants.ROUTE_ID, routeId);
+            resultIntent.putExtra(Constants.ROUTE_NAME, routeName);
+            resultIntent.putExtra(Constants.TRANSIT_TYPE, transitType);
+            resultIntent.putExtra(Constants.EXPIRATION_TIMESTAMP, expirationTimeStamp);
+
+            // build back stack
+            TaskStackBuilder stackBuilder = TaskStackBuilder.create(context);
+            stackBuilder.addNextIntentWithParentStack(resultIntent);
+            PendingIntent pendingIntent = stackBuilder.getPendingIntent(0, PendingIntent.FLAG_UPDATE_CURRENT);
+            mBuilder.setContentIntent(pendingIntent);
+
+        }
+
+        displayNotification(context, mBuilder);
+    }
+
+    private void displayNotification(Context context, NotificationCompat.Builder mBuilder) {
+        // dismiss notification on click
+        mBuilder.setAutoCancel(true);
+
+        // add sound / vibrate based on current user settings
+        mBuilder.setDefaults(Notification.DEFAULT_ALL);
+
+        // use timestamp to create unique notif ID
+        int notifId = (int) System.currentTimeMillis();
+
+        /*
+         * The first parameter is the notification id
+         * better don't give a literal here (right now we are giving a int literal)
+         * because using this id we can modify it later
+         * */
+        NotificationManager mNotifyMgr = (NotificationManager) context.getSystemService(NOTIFICATION_SERVICE);
+        if (mNotifyMgr != null) {
+            mNotifyMgr.notify(notifId, mBuilder.build());
+        }
+    }
+
+    private static void showNotificationExpiredMessage(Context context) {
+        Toast.makeText(context, R.string.notification_expired, Toast.LENGTH_LONG).show();
+    }
+
+    public static void onServiceAlertNotificationClick(Activity activity, Intent intent) {
```

Comment:
There really isn't any diffrence between these two methods other than what gets passed on the intent around which type of alert is expanded.  So maybe combine them and call them with a parameter?